test/unittests/test_user_infinite_loop.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+from unittest import mock
+import tap_zendesk
+from tap_zendesk.streams import Users
+from tap_zendesk.streams import STREAMS
+import singer
+
+
+class MockSearch:
+    def __init__(self):
+        self.count = 1001
+        self.updated_at = "test"
+        
+    def __iter__(self):
+        return (self for x in range(4))
+        
+    def search(self, test, updated_after, updated_before, type="user" ):
+        # For window size less than 2 return less than or equal to 1000 records and for larger window return greater than 1000 records
+        if (singer.strptime(updated_before) - singer.strptime(updated_after)).seconds < 2:
+            self.count = 999
+        else:
+            self.count = 1001
+        return self
+
+class TestUserSyncCheck(unittest.TestCase):
+
+    def test_many_records_in_one_seconds_for_user(self):
```

Comment:
On local env, this unit test is failing,

```
$ nosetests test/unittests/test_user_infinite_loop.py:TestUserSyncCheck.test_many_records_in_one_seconds_for_user
.
.
======================================================================
FAIL: Reproduce infinite looping behavior for Users stream when user have many record in single seconds
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/code/tap-zendesk/test/unittests/test_user_infinite_loop.py", line 36, in test_many_records_in_one_seconds_for_user
    self.assertEqual(str(e.exception), 'users - Unable to get all users within minimum window of a single second (2022-03-30T08:45:21Z), found 1001 users within this timestamp. Zendesk can only provide a maximum of 1000 users per request. See: https://de/
velop.zendesk.com/hc/en-us/articles/360022563994--BREAKING-New-Search-API-Result-Limits')
AssertionError: 'user[71 chars]022-04-14T08:45:20Z), found 1001 users within [176 chars]mits' != 'user[71 chars]022-03-30T08:45:21Z), found 1001 users within [176 chars]mits'
Diff is 801 characters long. Set self.maxDiff to None to see it.
-------------------- >> begin captured stdout << ---------------------
{"type": "STATE", "value": {"bookmarks": {"users": {"updated_at": "2022-04-14T08:45:20Z"}}}}

```

Reply:
We expect static datetime in the exception message, but the unit test is internally using the current datetime. So, we mocked now() with static datetime